baseline/services.py

```diff
@@ -481,25 +483,40 @@ def predict(self, tokens, **kwargs):
             examples = unfeaturized_examples
 
         outcomes = self.model.predict(examples)
-        return self.format_output(outcomes, tokens_batch=tokens_batch, label_field=label_field)
+        return self.format_output(outcomes, tokens_batch=tokens_batch, label_field=label_field, vectorized_examples=examples)
 
-    def format_output(self, predicted, tokens_batch=None, label_field='label', **kwargs):
+    def format_output(self, predicted, tokens_batch=None, label_field='label', vectorized_examples=None, **kwargs):
+        """This code got very messy dealing with BPE/WP outputs."""
         assert tokens_batch is not None
+        assert vectorized_examples is not None
         outputs = []
+        # Pick a random non-lengths key from the vectorized input, if one input was BPE'd they all had to be to stay aligned
+        key = [k for k in vectorized_examples if not k.endswith("_lengths")][0]
+        # Get the subword start signal from the vectorizer's tokenizer. If there is none we return None
+        subword_start = getattr(getattr(self.vectorizers[key], "tokenizer", None), "subword_start", None)
```

Comment:
i think this property would be better placed on the actual vectorizer (even if it just delegates to a sub-prop).  this is  a bit hectic here

Reply:
This property is now accessible on the vectorizer